controllers/argocd/route.go

```diff
@@ -366,3 +381,42 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 	}
 	return r.Client.Update(context.TODO(), route)
 }
+
+func shortenHostname(hostname string) (string, error) {
+	if hostname == "" {
+		return "", fmt.Errorf("hostname is empty")
+	}
```

Comment:
In this case, we should just `return "", nil`, e.g. correct me if I'm wrong, but the original code (which is calling this function) does not return an error if the hostname is empty.

Reply:
here at this [line](https://github.com/argoproj-labs/argocd-operator/blob/41356fb260e9158ae5a628d267eeab97e3ac6939/controllers/argocd/route.go#L229) we are  calling shortenHostname function when `cr.Spec.Server.Route.Enabled` is true , it means route is enable and having a hostname and hostname should not empty and we are overriding the hostname [here](https://github.com/argoproj-labs/argocd-operator/blob/41356fb260e9158ae5a628d267eeab97e3ac6939/controllers/argocd/route.go#L249) so I have returned the error instead of nil? I may be wrong in understanding this please do correct me

Reply:
For this case, I think we should preserve the behaviour of the original code: AFAICT, in the original code, if the host name was not specified, then the code would not return an error and would still create a Route with an empty host. Can you confirm? 

You could ask on #gitops-dev if this is desirable behaviour or not, but I'm not familiar enough with the expected behaviour in this case, so my gut is to not modify that property of the existing code.

Reply:
Yes @jgwest I tried this without hostname and it is not returning an error, I will keep the behaviour of existing code here, thank you